webrender/src/tiling.rs

```diff
@@ -1393,6 +1399,31 @@ impl PackedLayer {
     pub fn empty() -> PackedLayer {
         Default::default()
     }
+
+    pub fn set_transform(&mut self, transform: LayerToWorldTransform) {
+        self.transform = transform;
+        self.inv_transform = self.transform.inverse().unwrap();
+    }
+
+    pub fn set_rect(&mut self,
+                    local_rect: Option<LayerRect>,
+                    screen_rect: &DeviceIntRect,
+                    device_pixel_ratio: f32)
+                    -> Option<TransformedRect> {
+        let local_rect = match local_rect {
+            Some(rect) if !rect.is_empty() => rect,
+            _ => return None,
+        };
+
+        let xf_rect = TransformedRect::new(&local_rect, &self.transform, device_pixel_ratio);
+        if !xf_rect.bounding_rect.intersects(screen_rect) {
+            return None;
+        }
+
+        self.screen_vertices = xf_rect.vertices.clone();
+        self.local_clip_rect = local_rect;
+        Some(xf_rect)
```

Comment:
```screen_rect``` intersection is done in ```handle_pop_stacking_context```. Perhaps visibility should also be computed on a per-group basis as well. I believe the current design is holdover from when stacking contexts visibility was computed with a 1-to-1 stacking context and scroll layer pair.